sir_cake/seller/views.py

```diff
@@ -144,3 +148,42 @@ def _mark_order_as_shipped(request, order):
         order.status = Order.Status.SHIPPED
         order.save()
         messages.success(request, 'Order marked as shipped')
+
+
+def _order_context(order):
+    context = get_items_and_prices_and_order_sum(
+        BasketItem.objects.filter(order=order))
+    context['order'] = order
+    context['anonymous'] = order.buyer is None
+    return context
+
+
+class UserIsSellerMixin(UserPassesTestMixin):
+    def test_func(self):
+        return self.request.user.is_staff
+
+
+class PdfInvoiceView(UserIsSellerMixin, View):
+    def get(self, request, pk, *_, **__):
+        try:
+            order = Order.objects.get(id=pk)
+        except ObjectDoesNotExist:
+            return HttpResponseNotFound()
+
+        context = _order_context(order)
+        context['due_date'] = order.order_date + \
+            timezone.timedelta(days=settings.INVOICE_DUE_DATE_DAYS)
+        pdf = _render_pdf('seller/invoice.html', context, request)
+        return HttpResponse(pdf, content_type='application/pdf')
```

Comment:
I'm a bit confused, `_render_pdf` returns a `HttpResponse` or a `redirect`. How can this then be passed to another instance of `HttpResponse` here?